ksql-serde/src/main/java/io/confluent/ksql/serde/KsqlSerdeFactory.java

```diff
@@ -61,4 +106,17 @@ public boolean equals(final Object o) {
   public int hashCode() {
     return Objects.hash(format);
   }
+
+  private static Schema getSerializerSchema(final Schema schema, final KsqlConfig ksqlConfig) {
+    if (schema.fields().size() != 1) {
+      return schema;
+    }
+
+    if (ksqlConfig.getBoolean(KsqlConfig.KSQL_PERSIST_SINGLE_FIELD_IN_STRUCT)) {
```

Comment:
Just adding a comment with an internal monologue to reason about compatibility. If we have this config and later we switch to what we talked about offline (i.e. use the source for CSAS/CTAS statements) then does that cause any issues? 

- case config matches source: no problem
- case config true, source primitive: new serialization format will be primitive instead of struct! I think this is a breakage
- case config false, source struct: new serialization format will be struct instead of primitive! I think this is also a breakage

The question is, how bad are the breakages? With the "magic" deserialization KSQL won't break, but any downstream applications that were reading in the old format will no longer read values. Is that an okay breakage?